### PR TITLE
Modify generateSpend() to send change to a fresh confidential identity

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

**Comment by josecoll** [comment]
I don't understand the purpose of this change.

File: finance/src/main/kotlin/net/corda/finance/contracts/asset/Cash.kt [hunk 1]
```diff
@@ -327,7 +327,9 @@ class Cash : OnLedgerAsset<Currency, Cash.Commands, Cash.State>() {
             val totalAmount = payments.map { it.amount }.sumOrThrow()
             val cashSelection = CashSelection.getInstance({ services.jdbcSession().metaData })
             val acceptableCoins = cashSelection.unconsumedCashStatesForSpending(services, totalAmount, onlyFromParties, tx.notary, tx.lockId)
+            val changeIdentity = services.keyManagementService.freshKeyAndCert(services.myInfo.legalIdentityAndCert, revocationEnabled = false)
```
Comment (mikehearn) [inline finance/src/main/kotlin/net/corda/finance/contracts/asset/Cash.kt]: Can you add a comment explaining why there is a need for a fresh key and why revocationEnabled = false?

Comment (cioro) [reply finance/src/main/kotlin/net/corda/finance/contracts/asset/Cash.kt]: This code change breaks the build. services.myInfo does not contain legalIdentityAndCert anymore but legalIdentitiesAndCerts which List<PartyAndCertificate>.

**Comment by roger3cev** [comment]
I asked Ross to make these changes. This fixes a bug where by the change address for anonymous cash payments always included a `Party` as opposed to an `AnonymousParty`. Cheers

**Pull request merged**